model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,182 @@
+  repeated  SurveyQuestionAnswer answer_list = 5;
+}
+
+// The list of types a question can have that is used to determine how it is presented by the UI.
+enum QuestionType{
+  // The type of question is unknown.
+  QUESTION_TYPE_UNSPECIFIED=0;
+  // Corresponds to a question with multiple choices.
+  MULTIPLE_CHOICE = 1;
+
+  // Corresponds to a question with multiple choices and a free-text 'other' option.
+  MULTIPLE_CHOICE_WITH_OTHER = 2;
+
+  // Corresponds to a question which takes a free-text response.
+  FREE_FORM = 3;
+}
+
+// Corresponds to the semantic name given to each question.
+enum QuestionName{
+  // The question name is unknown.
+   QUESTION_NAME_UNSPECIFIED = 0;
+  // Corresponds to the name of the user type question.
+  QUESTION_USER_TYPE = 1;
+
+  //  Corresponds to the market fit question.
+  QUESTION_MARKET_FIT = 2;
+
+  // Corresponds to  the NPS question.
+  QUESTION_NPS = 3;
+
+  // Corresponds to  the open-ended question for a NPS promoter.
+  QUESTION_PROMOTER_FEEDBACK = 4;
+
+  // Corresponds to  the open-ended question for a NPS passive.
+  QUESTION_PASSIVE_FEEDBACK = 5;
+
+  // Corresponds to  the open-ended question for a NPS detractor.
+  QUESTION_DETRACTOR_FEEDBACK = 6;
+}
+
+// Represents the multiple choice answer available for a question.
+message SurveyQuestionAnswer{
+  // The ID of this answer option.
+  string question_answer_id = 1;
+
+  // The semantic name of the question this answer option is tied to.
+  QuestionName question_name = 2;
+
+  // The actual text value of the answer.
+  string answer = 3;
+}
+
+// Represents an answer that was selected by a user for a survey question.
+message SurveyUserAnswer{
+  //  The ID of the survey to which this answer is tied.
+  string  survey_id = 1;
+
+  // Semantic name of the question the response is tied to.
+  QuestionName question_name = 2;
+
+  // The value of the answer the user gave.
+  string user_answer = 3;
+}
+
+// Represents the different types of users who could be taking the survey.
+// Consists of answer options for the user type question.
+enum  UserTypeAnswer{
+  // User type unknown.
+  USER_TYPE_UNSPECIFIED =0;
+  // Corresponds to an unspecified or unknown type of user.
+  OTHER = 1;
+
+  // Corresponds to a user who is a teacher.
+  TEACHER = 2;
+
+  // Corresponds to a user who is a parent.
+  PARENT = 3;
+
+  // Corresponds to a user who is a learner.
+  LEARNER = 4;
+}
+
+// Represents general opinion of the user about the Oppia android app.
+// Consists of answer options for the market fit question.
+enum  MarketFitAnswer{
+  // Answer unknown.
+  MARKET_FIT_ANSWER_UNSPECIFIED =0;
+
+  // Corresponds to the answer for Very Disappointed.
+  VERY_DISAPPOINTED = 1;
+
+  // Corresponds to the answer for Disappointed.
+  DISAPPOINTED = 2;
+
+  // Corresponds to the answer for  Not Disappointed.
+  NOT_DISAPPONTED = 3;
+
+  // Corresponds to the answer for  N/A - Don't use Oppia Anymore.
+  NA_DONT_USE_OPPIA_ANYMORE = 4;
```

Comment:
'NA' is a bit ambiguous--perhaps 'NOT_APPLICABLE' would be clearer?

Also, wouldn't the context be "won't use Oppia anymore"? If they aren't using the app, they presumably wouldn't be filling out this survey.

Reply:
You're right about the context.
I was hopping to avoid extremely long names, but `NOT_APPLICABLE_WONT_USE_OPPIA_ANYMORE` is self documenting.